frontend/src/features/map_planning/layers/plant/actions.ts

```diff
@@ -0,0 +1,218 @@
+/**
+ * @module this module contains actions for the plant layer.
+ */
+import { createPlanting } from '../../api/createPlanting';
+import { deletePlanting } from '../../api/deletePlanting';
+import { movePlanting } from '../../api/movePlanting';
+import { transformPlanting } from '../../api/transformPlanting';
+import { Action, TrackedMapState } from '../../store/MapStoreTypes';
+import { PlantLayerObjectDto } from '@/bindings/definitions';
+
+export class CreatePlantAction
+  implements Action<Awaited<ReturnType<typeof createPlanting>>, boolean>
+{
+  private readonly _id: string;
+
+  constructor(private readonly _data: PlantLayerObjectDto) {
+    this._id = _data.id;
+  }
+
+  reverse() {
+    return new DeletePlantAction(this._id);
+  }
+
+  apply(state: TrackedMapState): TrackedMapState {
+    return {
+      ...state,
+      layers: {
+        ...state.layers,
+        Plant: {
+          ...state.layers.Plant,
+          objects: [
+            ...state.layers.Plant.objects,
+            {
+              ...this._data,
+              id: this._id,
+            },
+          ],
+        },
+      },
+    };
+  }
+
+  async execute(): Promise<Awaited<ReturnType<typeof createPlanting>>> {
+    return createPlanting({
+      ...this._data,
+      // TODO - get these values from the store.
+      map_id: 1,
+      plant_id: this._data.plantId,
+      id: this._id,
+    });
+  }
+}
+
+export class DeletePlantAction
+  implements Action<boolean, Awaited<ReturnType<typeof createPlanting>>>
+{
+  constructor(private readonly _id: string) {}
+
+  async execute(): Promise<boolean> {
+    return deletePlanting(this._id);
+  }
+
+  reverse(state: TrackedMapState) {
+    const plant = state.layers.Plant.objects.find((obj) => obj.id === this._id);
```

Comment:
I don't get how the reverse action works here.
I also can't test it because I don't know how to delete an object (Delete and Backspace key don't work).

Looks like the action can't be reversed because the object is no longer present in the state after deleting it.

Reply:
Ok, I got it now.
I am still not able to delete an object though.
Is this bug related to this PR?

Reply:
No, it is just not implemented in the UI yet. 
I will implement this as part of the bigger plant layer use case.

Reply:
Alright, I thought it was at some point but maybe I am mistaken.